### PR TITLE
soumyo/aws_ec2_transit_gateway_route_table_propagation

### PR DESCRIPTION
**Pull request by soumyo13**
Signed-off-by: Soumyodeep Karmakar <soumyo.k13@gmail.com>

### Description

It contains all the singular and plural resources of the above resource. 

### Issues Resolved
This is a new resource. There are no issues to be fixed.

### Check List
Please fill box or appropriate ([x]) or mark N/A.
- [x] New functionality includes integration tests/controls
- [x] New Terraform resources
- [x] Documentation provided or updated for resources 
- [x] All Integration Tests pass
- [x] All Unit Tests pass
- [x] `rake lint` passes
- [x] All commits have been signed-off for the Developer Certificate of Origin. See <https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco>

### CODE REVIEW

**Review by Nirbhay1997 — APPROVED** [review]

**Comment by IanMadd** [comment]
What's the difference between the two resources? Does one handle a single route table propagation and the other handles all route table propagations?

**Comment by ppradhan9** [comment]
closed the PR by mistake

File: docs/resources/aws_ec2_transit_gateway_route_table_propagations.md [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+| transit_gateway_attachment_ids | The ID of the attachment. | transit_gateway_attachment_id |
+| resource_ids | The ID of the resource. | resource_id |
+| resource_types | The type of resource. | resource_type |
+| states | The state of the resource. | state |
+
+## Examples
+
+### Ensure a transit gateway attachment id is available.
+    describe aws_ec2_transit_gateway_route_table_propagations(transit_gateway_route_table_id: 'TransitGatewayRouteTableId') do
+      its('transit_gateway_attachment_id') { should include 'TransitGatewayRouteTableId' }
```
Comment (Nirbhay1997) [inline docs/resources/aws_ec2_transit_gateway_route_table_propagations.md:45]:
```suggestion
      its('transit_gateway_attachment_ids') { should include 'TransitGatewayRouteTableId' }
```

Comment (soumyo13) [reply docs/resources/aws_ec2_transit_gateway_route_table_propagations.md]: Done thanks

**Comment by soumyo13** [comment]
> What's the difference between the two resources? Does one handle a single route table propagation and the other handles all route table propagations?

Yeah. These are 2 different resources. One is singular and other is the plural.

**Comment by soumyo13** [comment]
Now I have modified the singular resource with more filter options. Now the user has to audit with both transit_gateway_route_table_id and transit_gateway_attachment_id.

**Review by IanMadd — APPROVED** [review]

**Pull request merged**